cql3/query_processor.hh

```diff
@@ -123,6 +123,7 @@ private:
     service::migration_notifier& _mnotifier;
     service::migration_manager& _mm;
     const cql_config& _cql_config;
+    bool _tracing_required = false;
```

Comment:
Unfortunately, this approach with `_tracing_required` on `query_processor` is not correct. For example, it will be only set if the client ever sends a `PREPARE` request. There is only one `query_processor` per `transport/server.cc`, so `_tracing_required` would be incorrectly reused between different sessions.

The approach you should take is to reuse the machinery behind the "classic" tracing. There it works this way: if the user requested "classic" tracing on a request, it builds non-empty `trace_state_ptr`. Otherwise, an "empty" `trace_state_ptr` is passed around.

Here it figures out what type of tracing should be used: https://github.com/scylladb/scylla/blob/6e2d29300c79a8f5af76641972e5be7d9d381bf9/transport/server.cc#L601-L608

Here it actually builds a `trace_state_ptr`: https://github.com/scylladb/scylla/blob/6e2d29300c79a8f5af76641972e5be7d9d381bf9/transport/server.cc#L388

Your code should correctly fill out `trace_state_ptr` based on whether the client has `OPENTELEMETRY_TRACING`.